app/controllers/application_controller.rb

```diff
@@ -929,6 +930,23 @@ def should_enforce_2fa?
       !current_user.has_any_second_factor_methods_enabled?
   end
 
+  def redirect_to_profile_if_required
+    return if request.format.json? && is_api?
+    return if !current_user
+    return if !current_user.needs_required_fields_check?
+
+    if current_user.populated_required_custom_fields?
```

Comment:
The crux here is we can't guarantee this will work because by design the version number and the fields might not agree. In particular if an admin creates a new required field (bumps the version) and then changes it to be not required (or deletes it). The user's version will now be out of date, but they have no missing required fields.

> Otherwise we have to check this every page load, when we already know if they need the check or not in a cheaper call on line 936 above.

The L:936 one is a short-circuit, so it's only every page load for a user with an out-of-date version (i.e. not in the hot path.)

Reply:
Okay cool, sounds reasonable to me :+1: